### ISSUE TITLE
Error enrol users in scheduled task

### ISSUE THREAD

**Issue by dasistwas** [Wunderbyte-GmbH/moodle-mod_booking#185]
Moodle 3.11.6
Booking v7.0.20.02, 2021111604
```
Execute scheduled task: Booking (mod_booking\task\enrol_bookedusers_tocourse)
... started 03:54:02. Current memory use 6.3MB.
WARNING: Failed to get booking instance from option id: 44
... used 2 dbqueries
... used 0.042197942733765 seconds
Scheduled task failed: Booking (mod_booking\task\enrol_bookedusers_tocourse),Can't find data record in database.
Debug info:
SELECT cm.*, m.name, md.name AS modname 
              FROM {course_modules} cm
                   JOIN {modules} md ON md.id = cm.module
                   JOIN {booking} m ON m.id = cm.instance
                   
             WHERE cm.id = :cmid AND md.name = :modulename
                   
[array (
  'cmid' => NULL,
  'modulename' => 'booking',
)]
Backtrace:
* line 1262 of /lib/datalib.php: call to moodle_database->get_record_sql()
* line 87 of /mod/booking/classes/booking.php: call to get_coursemodule_from_id()
* line 125 of /mod/booking/classes/booking_option.php: call to mod_booking\booking->__construct()
* line 51 of /mod/booking/classes/task/enrol_bookedusers_tocourse.php: call to mod_booking\booking_option->__construct()
* line 253 of /lib/cronlib.php: call to mod_booking\task\enrol_bookedusers_tocourse->execute()
* line 120 of /lib/cronlib.php: call to cron_run_inner_scheduled_task()
* line 73 of /lib/cronlib.php: call to cron_run_scheduled_tasks()
* line 178 of /admin/cli/cron.php: call to cron_run()

```

**Comment by dasistwas** [comment]
```
Execute scheduled task: Booking (mod_booking\task\enrol_bookedusers_tocourse)
... started 15:11:10. Current memory use 3.9MB.
--------------------------------
SELECT id, bookingid FROM mdl_booking_options WHERE enrolmentstatus < 1 AND coursestarttime < $1
[array (
  0 => 1649509870,
)]
--------------------------------
Query took: 0.0014619827270508 seconds.
--------------------------------
WARNING: Failed to get booking instance from option id: 44
PHP Notice:  Undefined variable: cm in mod/booking/classes/task/enrol_bookedusers_tocourse.php on line 51
PHP Notice:  Trying to get property 'id' of non-object in mod/booking/classes/task/enrol_bookedusers_tocourse.php on line 51
--------------------------------
SELECT cm.*, m.name, md.name AS modname
              FROM mdl_course_modules cm
                   JOIN mdl_modules md ON md.id = cm.module
                   JOIN mdl_booking m ON m.id = cm.instance

             WHERE cm.id = $1 AND md.name = $2

[array (
  0 => NULL,
  1 => 'booking',
)]
--------------------------------
Query took: 0.0023159980773926 seconds.
--------------------------------
... used 2 dbqueries
... used 0.040402889251709 seconds
Scheduled task failed: Booking (mod_booking\task\enrol_bookedusers_tocourse),Can't find data record in database. (SELECT cm.*, m.name, md.name AS modname
              FROM {course_modules} cm
                   JOIN {modules} md ON md.id = cm.module
                   JOIN {booking} m ON m.id = cm.instance

             WHERE cm.id = :cmid AND md.name = :modulename

[array (
  'cmid' => NULL,
  'modulename' => 'booking',
)])
Debug info:
SELECT cm.*, m.name, md.name AS modname
              FROM {course_modules} cm
                   JOIN {modules} md ON md.id = cm.module
                   JOIN {booking} m ON m.id = cm.instance

             WHERE cm.id = :cmid AND md.name = :modulename

[array (
  'cmid' => NULL,
  'modulename' => 'booking',
)]
Backtrace:
* line 1262 of /lib/datalib.php: call to moodle_database->get_record_sql()
* line 87 of /mod/booking/classes/booking.php: call to get_coursemodule_from_id()
* line 125 of /mod/booking/classes/booking_option.php: call to mod_booking\booking->__construct()
* line 51 of /mod/booking/classes/task/enrol_bookedusers_tocourse.php: call to mod_booking\booking_option->__construct()
* line 253 of /lib/cronlib.php: call to mod_booking\task\enrol_bookedusers_tocourse->execute()
* line 167 of /admin/cli/scheduled_task.php: call to cron_run_inner_scheduled_task()

--------------------------------
INSERT INTO mdl_task_log (type,component,classname,userid,timestart,timeend,dbreads,dbwrites,result,output,hostname,pid) VALUES($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12) RETURNING id
[array (
  'type' => 0,
  'component' => 'mod_booking',
  'classname' => 'mod_booking\\task\\enrol_bookedusers_tocourse',
  'userid' => 0,
  'timestart' => 1649509870.9538701,

```

**Comment by bernhard-wunderbyte** [comment]
Fixed in Version 7.0.20.03 (2021111605).
